cub/cub/detail/choose_offset.cuh

```diff
@@ -56,6 +56,36 @@ struct ChooseOffsetT
                                          unsigned long long>::type;
 };
 
+template <typename T,typename = void>
+struct is_iterator : std::false_type {};
+template <typename T>
+struct is_iterator<T,std::void_t<typename std::iterator_traits<T>::iterator_category>> : std::true_type {};
```

Comment:
important: `std::void_t` is C++17 extension. CUB has to support C++11.

Reply:
Also noted.

Reply:
You could always use `cuda::std::__void_t` 😈

Reply:
I mean in libcu++ we have a whole lot of code to determine things like this. 

So maybe you could use either `__is_cpp17_input_iterator` or `__has_iterator_category`